src/EditorFeatures/CSharpTest/EditAndContinue/ActiveStatementTests.cs

```diff
@@ -8942,6 +8942,130 @@ static void F(object o1, object o2)
 
         #endregion
 
+        #region Nullable
+
+        [Fact]
+        public void ChangeLocalNullableToNonNullable()
+        {
+            string src1 = @"
+class C
+{
+    static void F()
+    {
+        <AS:0>string? s = ""a"";</AS:0>
+    }
+}
+";
+            string src2 = @"
+class C
+{
+    static void F()
+    {
+        <AS:0>string s = ""a"";</AS:0>
+    }
+}
+";
+            var edits = GetTopEdits(src1, src2);
+            var active = GetActiveStatements(src1, src2);
+
+            edits.VerifyRudeDiagnostics(active);
+        }
+
+        [Fact]
+        public void ChangeLocalNonNullableToNullable()
+        {
+            string src1 = @"
+class C
+{
+    static void F()
+    {
+        <AS:0>string s = ""a"";</AS:0>
+    }
+}
+";
+            string src2 = @"
+class C
+{
+    static void F()
+    {
+        <AS:0>string? s = ""a"";</AS:0>
+    }
+}
+";
+            var edits = GetTopEdits(src1, src2);
+            var active = GetActiveStatements(src1, src2);
+
+            edits.VerifyRudeDiagnostics(active);
+        }
+
+        [Fact]
+        public void ChangeArrayOfNullableToArrayOfNonNullable()
```

Comment:
>ChangeArrayOfNullableToArrayOfNonNullable [](start = 20, length = 41)

The foreach tests do not test anything interesting as they are written. We already tested changing the type of a local variable definition above. 

What would be interesting to test in `foreach`  loop is when the active statement is inside the loop (e.g. in `{`) and the type of `arr` is changed at that point (either explicitly or implicitly via non-nullable inference). We also don't need to test all combinations of nullable to non-nullable in that case. One is enough.

Reply:
It would be better to add such `foreach` tests to the `ForEach Statement` section since `foreach` is the major feature EnC-wise and nullability is just one of the ways the type of the iterator may change.

---
In reply to: [301274368](https://github.com/dotnet/roslyn/pull/36770#discussion_r301274368) [](ancestors = 301274368)